cmd/manager/main.go

```diff
@@ -65,128 +69,218 @@ const (
+	cmd.Flags().Bool(
 		operator.EnableWebhookFlag,
 		false,
 		"Enables a validating webhook server in the operator process.",
 	)
-	Cmd.Flags().Bool(
+	cmd.Flags().Bool(
 		operator.ManageWebhookCertsFlag,
 		true,
 		"Enables automatic certificates management for the webhook. The Secret and the ValidatingWebhookConfiguration must be created before running the operator",
 	)
-	Cmd.Flags().Int(
+	cmd.Flags().Int(
 		operator.MaxConcurrentReconcilesFlag,
 		3,
 		"Sets maximum number of concurrent reconciles per controller (Elasticsearch, Kibana, Apm Server etc). Affects the ability of the operator to process changes concurrently.",
 	)
-	Cmd.Flags().Int(
+	cmd.Flags().Int(
 		operator.MetricsPortFlag,
 		DefaultMetricPort,
 		"Port to use for exposing metrics in the Prometheus format (set 0 to disable)",
 	)
-	Cmd.Flags().StringSlice(
+	cmd.Flags().StringSlice(
 		operator.NamespacesFlag,
 		nil,
 		"comma-separated list of namespaces in which this operator should manage resources (defaults to all namespaces)",
 	)
-	Cmd.Flags().String(
+	cmd.Flags().String(
 		operator.OperatorNamespaceFlag,
 		"",
 		"K8s namespace the operator runs in",
 	)
-	Cmd.Flags().String(
+	cmd.Flags().String(
 		operator.WebhookCertDirFlag,
 		// this is controller-runtime's own default, copied here for making the default explicit when using `--help`
 		filepath.Join(os.TempDir(), "k8s-webhook-server", "serving-certs"),
 		"Path to the directory that contains the webhook server key and certificate",
 	)
-	Cmd.Flags().String(
+	cmd.Flags().String(
 		operator.WebhookSecretFlag,
 		"",
 		fmt.Sprintf("K8s secret mounted into the path designated by %s to be used for webhook certificates", operator.WebhookCertDirFlag),
 	)
-	Cmd.Flags().String(
+	cmd.Flags().String(
 		operator.WebhookConfigurationNameFlag,
 		DefaultWebhookConfigurationName,
 		fmt.Sprintf("Name of the Kubernetes ValidatingWebhookConfiguration resource (defaults to %s). Only used when enable-webhook is true.", DefaultWebhookConfigurationName),
 	)
 
-	// enable using dashed notation in flags and underscores in env
-	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	logconf.BindFlags(cmd.Flags())
 
-	if err := viper.BindPFlags(Cmd.Flags()); err != nil {
-		log.Error(err, "Unexpected error while binding flags")
-		os.Exit(1)
+	_ = cmd.MarkFlagFilename(operator.ConfigFlag)
+
+	return cmd
+}
+
+func doRun(_ *cobra.Command, _ []string) error {
+	signalChan := signals.SetupSignalHandler()
+
+	// no config file to watch so start the operator
+	if configFile == "" {
+		return startOperator(signalChan)
+	}
+
+	// receive config file update events over a channel
+	confUpdateChan := make(chan struct{}, 1)
+
+	// start watching the file for changes unless the DisableConfigWatch flag is set
+	if !viper.GetBool(operator.DisableConfigWatch) {
+		log.Info("Watching config file for changes", "file", configFile)
+		viper.WatchConfig()
+		viper.OnConfigChange(func(evt fsnotify.Event) {
+			switch evt.Op {
```

Comment:
An event may contain several operations, e.g. `fsnotify.Write | fsnotify.Chmod` (bitwise OR)
```suggestion
			if evt.Op&fsnotify.Write == fsnotify.Write || evt.Op&fsnotify.Create == fsnotify.Create {
				confUpdateChan <- struct{}{}
			}
```